test/new_relic/agent/method_visibility_test.rb

```diff
@@ -67,19 +61,11 @@ def setup
   end
 
   %w| public private protected |.each do |visibility|
-    define_method "test_should_preserve_visibility_of_#{visibility}_traced_method" do
-      assert @instance.send("#{visibility}_methods").map{|s|s.to_sym}.include?(:"#{visibility}_method!"), "Method #{visibility}_method should be #{visibility}"
-    end
-
     define_method "test_should_preserve_visibility_of_#{visibility}_traced_transaction" do
       assert @instance.send("#{visibility}_methods").map{|s|s.to_sym}.include?(:"#{visibility}_transaction!"), "Transcation #{visibility}_transaction should be #{visibility}"
     end
   end
 
-  def test_tracing_non_public_methods_doesnt_add_public_methods
```

Comment:
I remember you mentioning in pairing that prepend makes things public, is that correct? I'm guessing that's why this test was removed. I'm curious what kind of impacts having it be public due to prepend would have for customers?

Reply:
I'm going to revisit this and add these tests back in. Whatever I was trying last week wasn't working and I decided to shelve it temporarily.

I misspoke, basically, since I tend to forget the fundamental of private methods when mixins get involved. Private methods aren't totally isolated to the class they're defined in; they just can't be called with an explicit receiver. This, for example, works:

    module M
      private
      def fruit
        "pine#{super}"
      end
    end

    class A
      prepend M

      def eat_something
        puts "Mmm, #{fruit}."
      end

      private
      def fruit
        'apple'
      end
    end

    A.new.eat_something  # "Mmm, pineapple."